Pagination // update lots of visuals

joe-s-avaya opened the pull request:
[Link to updated Pagination stories](https://deploy-preview-533--neo-react-library-storybook.netlify.app/?path=/docs/components-pagination--docs)
[Link to Figma reference](https://www.figma.com/design/SQHQlIhFHWCjngHZOnK8MH/Web-Components-2.0?node-id=6170-2217&node-type=FRAME&t=1ikRcQjuIKuF9VyB-0)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [x] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`


`@avaya-dux/dux-design`: Per our discussions, review the updates that discussed and let me know if there is anything missing/wrong.

- font colors
- appropriate use of commas (eg, "10,000" "100")
- background colors
- hover colors
- light vs dark mode colors
- width and height of buttons
- gap between elements
- vertically centered ellipses icon

`@avaya-dux/dux-devs`: review for functionality, I will be doing a PR for cleanup of the CSS after Design signs off and this PR is merged

- note that I'm using `*.toLocaleString()` in a bunch of places per Matt and Justin's request
- I'm now using the `ellipses-horizontal` icon instead of three periods
- we are now disabling the active page button and overwriting "disabled" styles per Matt and Justin's request, this meant that I had to update several unit tests to account for this change


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
	- Enhanced pagination display with localized formatting for improved readability.
	- Introduced a new `NavigationMenuButton` component for better code maintainability.

- **Bug Fixes**
	- Disabled navigation buttons for the current page to prevent unnecessary interactions.

- **Style**
	- Updated styles for pagination components, including new button styles, hover effects, and improved spacing for better visual consistency and usability.
	- Introduced grid layout for pagination when display size exceeds 1000px.
	- Adjusted heights of input elements for enhanced usability.
	- Improved visual alignment of pagination buttons with new color settings and hover effects.

- **Chores**
	- Updated the `@avaya/neo` package version for potential improvements and bug fixes.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

enrique-prado commented on src/components/Pagination/Pagination_shim.css:
```diff
@@ -27,17 +72,46 @@
+	margin: 0;
+
+	& li {
+		height: 32px;
```
I remember that Matt said that the pagination elements needed to be 24px in height but the Figma file incorrectly specified them as 32px.  He has supposed to correct that.  Is this 32px height correct?

matthewAvaya replied:
> I remember that Matt said that the pagination elements needed to be 24px in height but the Figma file incorrectly specified them as 32px. He has supposed to correct that. Is this 32px height correct?

I misquoted and confirmed it was 32px. @joe-s-avaya has already handled the changes. Thanks!

enrique-prado reviewed (approved):
Just the one question about the height of elements.

Justin-Avaya commented:
**Issue 1 of 3**
Page button hover colors - we have an inconsistency with the page button hover states vs the < and > button hover states. I don't believe this is a blocker and is most likely out of scope. I will bring this to the teams attention in our next daily standup.

<hr>

**Issue 2 of 3**
Dark mode colors are not displaying correctly for me. See example.
<img width="1301" alt="Screenshot 2024-09-13 at 12 38 40 PM" src="https://github.com/user-attachments/assets/b8250989-5f8e-406d-bc63-c84e18f24970">

<hr>

**Issue 3 of 3**
Not sure how I managed to do this, but I did. Unfortunately, can't replicate. See example.
<img width="983" alt="Screenshot 2024-09-13 at 12 50 31 PM" src="https://github.com/user-attachments/assets/68fdf55f-8a97-45ed-aacf-818f1c76e369">

<hr>

**Additional comments**
- As expected, the commas have made the buttons wider. Not a blocker but this needs to be noted, maybe more so for design.
- Selecting different pages still causes the pages section to jump around. However, if i'm to understand correctly, this is going to be fixed elsewhere at another time.
- Everything else looks great.

joe-s-avaya commented:
@Justin-Avaya 
> **Issue 1 of 3** Page button hover colors - we have an inconsistency with the page button hover states vs the < and > button hover states. I don't believe this is a blocker and is most likely out of scope. I will bring this to the teams attention in our next daily standup.

I've [created a ticket for this](https://avaya.atlassian.net/browse/NEO-2395). Will have to be done in the future as it's own ticket.


> **Issue 2 of 3** Dark mode colors are not displaying correctly for me. 

Updated.


> **Issue 3 of 3** Not sure how I managed to do this, but I did. Unfortunately, can't replicate. 

Could not reproduce. No work to be done. 


**OTHER**: updated selected page to be semi-bold.

joe-s-avaya commented:
@avaya-dux/dux-devs the code has not changed significantly, but it has changed. You have both already approved, but you may want to take a quick glance at the changes.
